src/libs/ReportScrollManager/index.native.js

```diff
@@ -4,13 +4,33 @@ import React from 'react';
 // to the original ref.
 const flatListRef = React.createRef();
 
+// A reference to the last index required to scroll to, for async layout change handler.
+let lastIndex;
+
 /**
  * Scroll to the provided index.
  *
  * @param {Object} index
  */
 function scrollToIndex(index) {
     flatListRef.current.scrollToIndex(index);
+    lastIndex = index;
+}
+
+/**
+ * Layout Change the latest indexes set
+ * The is a fallback handler for asyncrounius scrolling and layout changes.
+ */
+function layoutChange() {
+    if (!flatListRef.current) {
+        return;
+    }
+    if (lastIndex === undefined) {
+        return;
+    }
+
+    flatListRef.current.scrollToIndex(lastIndex);
+    lastIndex = undefined;
 }
 
```

Comment:
Should we update `lastIndex` when `scrollToBottom` is called?

Reply:
added lastIndex = 0;